src/App.tsx

```diff
@@ -1,24 +1,193 @@
-/* eslint-disable max-len */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useState } from 'react';
 import { UserWarning } from './UserWarning';
+import { Todo } from './types/Todo';
+import { SortBy } from './types/SortBy';
+import { addTodo, deleteTodo } from './api/todos';
+import { Todos } from './components/Todos';
+import { useGetTodos } from './hooks';
 
-const USER_ID = 0;
+const USER_ID = 11361;
 
 export const App: React.FC = () => {
+  const [sortBy, setSortBy] = useState<SortBy>(SortBy.all);
+  const [selectedTodo, setSelectedTodo] = useState<number[]>([]);
+  const [inputValue, setInputValue] = useState<string>('');
+  const [isLoadingTodo, setIsLoadingTodo] = useState(false);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const {
+    isLoading,
+    todos,
+    errorMessage,
+    todosNotCompleted,
+    handleError,
+    handleIsTodoDeleted,
+  } = useGetTodos(USER_ID, tempTodo);
+
+  const deleteOneTodo = async (todoId: number) => {
+    try {
+      await deleteTodo(USER_ID, todoId);
+    } catch (error) {
+      handleError('Unable to delete a todo');
+    }
+
+    handleIsTodoDeleted(true);
```

Comment:
The todo could be removed right after the deleteTodo function resolves. You probably don't need this setState.

Reply:
this function change "isTodoDeleted" wich is declared in hooks.ts. When user delete todo, then isTodoDeleted is changed to true. 
Later the getTodos function is called and after getting the data from the server "isLoading" is changed to true. Thanks to this, the "loader" is displayed on the removed todos until the server responds